modules/doc/content/help/development/VSCode.md

```diff
@@ -0,0 +1,84 @@
+# Setup VSCode for MOOSE code and input file development
+
+!media framework/vscode/autocomplete2.gif
+       style=width:300px;padding-left:20px;float:right;
+       caption=VSCode autocompletion in MOOSE input files.
+
+!media framework/vscode/autocomplete1.gif
+       style=width:300px;padding-left:20px;float:right;clear:right;
+       caption=Autocompletion for MOOSE input files.
+
+!media framework/vscode/outline.gif
+       style=width:300px;padding-left:20px;float:right;clear:right;
+       caption=Outlines for MOOSE input files.
+
+!media framework/vscode/format_document.gif
+       style=width:300px;padding-left:20px;float:right;clear:right;
+       caption=MOOSE input file formatting.
+
+[VSCode](https://code.visualstudio.com/) is a text editor developed by Microsoft with a flexible plugin structure.
+VSCode superseedes Atom. To customize VSCode for MOOSE development the [MOOSE Language Support](https://marketplace.visualstudio.com/items?itemName=DanielSchwen.moose-language-support) extension has been developed.
+
+## Setting up VSCode
+
+Installation packages can be downloaded for Mac, Linux, and Windows operating systems from the
+[VSCode website](https://code.visualstudio.com/). The editor has an automatic update system for both the core editor as
+well as the installed extensions.
+
+### First Steps
+
+- On macOS +Start VSCode+ for the first time press +`F1`+ (or +`Cmd-Shift-P`+) and type `Shell Command: Install 'code' command in PATH`. Press enter. This activates the
+  `code` shell command for use from a terminal.
+- On Linux the shell command is automatically installed.
+- +Close VSCode+. From now on we will _only_ start it from the command line using the `code`
+  terminal command ensuring that VSCode sees the full MOOSE build environment.
+
+## Important commands
+
+- +`Cmd-Shift-P`+ opens the command palette. Every available Atom command can be accessed by typing a
+  few letters here. The dropdown list shows the keyboard shortcuts.
+- +`Cmd-P`+ opens a file anywhere in the current project tree (i.e. below the directory in which you
+  issued the `atom .` command. No need to know the precise path or even the precise spelling of the
+  filename!
+
+## Extensions
+
+
+The following extensions should be installed to effectively develop MOOSE based codes and edit MOOSE
+input files using Atom
+
+- [MOOSE Language Support](https://marketplace.visualstudio.com/items?itemName=DanielSchwen.moose-language-support): Syntax highlighting, automatic
+  indentation, document outlines, mouse hover info, formatting, and auto-completion for MOOSE input files, C++ code snippets for all MOOSE systems, and
+  highlighting of select MOOSE C++ types.
+- [C/C++ Extension Pack](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools-extension-pack) installs a set of extensions to support C++ code navigation, syntax highlighting, etc.
+
```

Comment:
There is also the switch header source extension https://marketplace.visualstudio.com/items?itemName=ryzngard.vscode-header-source

Reply:
I'm looking at the old Atom page. It seems that we are still missing make-runner, clang-format and rtags here, but vscode can do all that already. Perhaps we should list those extensions here for completeness.

Reply:
Yeah, I chose not to list stuff that's already built in. For new users that aren't coming from Atom this would just be confusing. By the way, som people swear by the clangd extension from LLVM for better code completion and navigation. I have yet to test it (as it is a bit more complicated than teh intellisense stuff - you need to (re)build the `compile-commands.json` file). We can amend this page as we get more experienced with VSCode.